File: ARCHISOFT_topoalign/Pick Filters/TopoPickFilter.vb
Public Class TopoPickFilter
    Implements ISelectionFilter

    Public Function AllowElement(elem As Element) As Boolean Implements ISelectionFilter.AllowElement
        If elem.Category Is Nothing Then
            Return False
        End If
        If elem.Category.Id.IntegerValue = CInt(BuiltInCategory.OST_Topography) Then
            Return True
        End If
        Return False
        'Return (elem.Category.Id.IntegerValue.Equals(CInt(BuiltInCategory.OST_Topography)))
    End Function

    Public Function AllowReference(reference As Reference, position As XYZ) As Boolean Implements ISelectionFilter.AllowReference
        Return False
    End Function
End Class

File: TopoAlign/PickFilters/TopoPickFilter.cs
using Autodesk.Revit.DB;
using Autodesk.Revit.UI.Selection;

namespace TopoAlign;

public class TopoPickFilter : ISelectionFilter
{
    public bool AllowElement(Element elem)
    {
        if (elem.Category is null)
        {
            return false;
        }

#if REVIT2024_OR_GREATER
        if (elem.Category.Id.Value == (int)BuiltInCategory.OST_Toposolid)
        {
            return true;
        }
#else
        if (elem.Category.Id.IntegerValue == (int)BuiltInCategory.OST_Topography)
        {
            return true;
        }
#endif

        return false;
        // Return (elem.Category.Id.IntegerValue.Equals(CInt(BuiltInCategory.OST_Topography)))
    }

    public bool AllowReference(Reference reference, XYZ position)
    {
        return false;
    }
}
